core/array/hash_spec.rb

```diff
@@ -11,29 +11,33 @@
     end
   end
 
-  it "properly handles recursive arrays" do
-    empty = ArraySpecs.empty_recursive_array
-    empty.hash.should be_kind_of(Fixnum)
+  ruby_bug "#", "1.8.6.277" do
+    it "properly handles recursive arrays" do
+      empty = ArraySpecs.empty_recursive_array
+      lambda { empty.hash }.should_not raise_error
 
-    array = ArraySpecs.recursive_array
-    array.hash.should be_kind_of(Fixnum)
+      array = ArraySpecs.recursive_array
+      lambda { empty.hash }.should_not raise_error
+    end
   end
 
-  it "returns the same hash for equal recursive arrays" do
-    rec = []; rec << rec
-    rec.hash.should == [rec].hash
-    rec.hash.should == [[rec]].hash
-    # This is because rec.eql?([[rec]])
-    # Remember that if two objects are eql?
-    # then the need to have the same hash
-    # Check the Array#eql? specs!
-  end
+  ruby_bug "redmine #1852", "1.9.1" do
+    it "returns the same hash for equal recursive arrays" do
+      rec = []; rec << rec
+      rec.hash.should == [rec].hash
+      rec.hash.should == [[rec]].hash
+      # This is because rec.eql?([[rec]])
+      # Remember that if two objects are eql?
+      # then the need to have the same hash
+      # Check the Array#eql? specs!
+    end
 
-  it "returns the same hash for equal recursive arrays through hashes" do
-    h = {} ; rec = [h] ; h[:x] = rec
-    rec.hash.should == [h].hash
-    rec.hash.should == [{:x => rec}].hash
-    # Like above, this is because rec.eql?([{:x => rec}])
+    it "returns the same hash for equal recursive arrays through hashes" do
+      h = {} ; rec = [h] ; h[:x] = rec
+      rec.hash.should == [h].hash
+      rec.hash.should == [{:x => rec}].hash
+      # Like above, this is because rec.eql?([{:x => rec}])
+    end
   end
```

Comment:
I think the changes in this file are just the result of a bad merge. They just have additional `ruby_bug` but on unsupported versions so they were removed.